### ISSUE TITLE
problems with concurrent requests to same device

### ISSUE THREAD

**Issue by palsbo** [njh/node-owfs#11]
Hi
I am loopeing thru a dir request in order to analyse the devices. 
if a DS1820 I wnat to read the temperature which goes well
## if a DS2438 I want to read Temperature, VAD and VDD - this will fail with the following:

/volume2/server/owfstest/node_modules/owfs/build/base/convert.js:19
        result = messageToUse.payload.replace(new RegExp(" ", "g"), "");
                                      ^
TypeError: Cannot call method 'replace' of undefined
    at /volume2/server/owfstest/node_modules/owfs/build/base/convert.js:19:39
    at callbackOnce (/volume2/server/owfstest/node_modules/owfs/build/base/communication.js:23:9)
    at Socket.<anonymous> (/volume2/server/owfstest/node_modules/owfs/build/base/communication.js:32:14)
    at Socket.emit (events.js:117:20)
    at _stream_readable.js:943:16
##     at process._tickCallback (node.js:419:13)

What can I do?

I tried to make the read for VAD inside the result for Temerature read result and the read for VDD inside the read for VAD result in order to prevent multible outstanding requests for one device, but it still fails

**Comment by njh** [comment]
Hello,

I have found that owserver doesn't cope will with multiple concurrent reads.
To solve this problem in my code, I originally used the eachSeries function from async:
http://caolan.github.io/async/docs.html#eachSeries

And later switch to using promises from Bluebird:
https://github.com/njh/node-red-contrib-owfs/blob/master/owfs.js

This makes the code synchronous, and only allows a single read at a time - and I stopped seeing errors.

nick.
